homeassistant/components/thethingsnetwork/entity.py

```diff
@@ -0,0 +1,203 @@
+"""Support for The Things Network entities."""
+
+from abc import ABC, abstractmethod
+import logging
+from typing import TYPE_CHECKING, Optional
+
+from ttn_client import TTNBaseValue, TTNSensorValue
+
+from homeassistant.core import callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+if TYPE_CHECKING:
+    from homeassistant.config_entries import ConfigEntry
+
+    from .coordinator import TTNCoordinator
+
+from .const import (
+    CONF_APP_ID,
+    OPTIONS_DEVICE_NAME,
+    OPTIONS_FIELD_ICON,
+    OPTIONS_FIELD_NAME,
+    OPTIONS_FIELD_PICTURE,
+    OPTIONS_FIELD_UNIT_MEASUREMENT,
+)
+from .entry_settings import TTN_EntrySettings
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TTN_Entity(CoordinatorEntity, Entity, ABC):
+    """Representation of a The Things Network Data Storage sensor."""
+
+    @staticmethod
+    def get_unique_id(device_id: str, field_id: str) -> str:
+        """Get unique_id which is derived from device_id and field_id."""
+        return f"{device_id}_{field_id}"
```

Comment:
Not sure why this has to be a static one (I have not encountered the use yet so maybe I will in a few moments)

Reply:
it is used in sensor.py to check if the entity already exists.

Reply:
Yes, but why don't we just directly set this in the constructor?

Reply:
ok, removing as I avoided the unique_id outside the constructor.